### PR TITLE
Doc from openstack admin guide

### PR DESCRIPTION
**Pull request by seunghun1ee**
Moved general operation documentations from Openstack admin guide.
Removed customer specific templatings and tried to generalise.
Updated old contents
- Update hardware inspection with Cardiff tool to ADVise tool
- Remove Ceph-ansible section of Ceph management
- Updated Nova client command to Openstack client commands

Please let me know if there are more contents that need to be updated.

Related PR on Openstack Admin Guide https://github.com/stackhpc/openstack-admin-guide/pull/53

### CODE REVIEW

File: doc/source/operations/baremetal-node-management.rst
Comment (seunghun1ee) [inline doc/source/operations/baremetal-node-management.rst]: Question: Can those details mentioned in TODOs be generalised as well? Or should we keep this in admin guide?

File: doc/source/operations/baremetal-node-management.rst
Comment (seunghun1ee) [inline doc/source/operations/baremetal-node-management.rst]: Question: Do we have standard way of doing this now?

File: doc/source/operations/baremetal-node-management.rst
Comment (Alex-Welsh) [reply doc/source/operations/baremetal-node-management.rst]: Either would work. General instructions could describe the most common setups e.g. how to trigger a PXE boot on the most common hardware, when describing the kayobe inventory, it could give a few example paths for where those file might be. It won't be as precise as custom per-deployment instructions but that is quite high-effort

File: doc/source/operations/ceph-management.rst
Comment (Alex-Welsh) [inline doc/source/operations/ceph-management.rst:1]: I we should have a link to the cephadm.rst page somewhere on this page. There is a lot of overlap but I do think they should be separate pages because this one is operations-focussed, the other is deployment-focussed

File: doc/source/operations/ceph-management.rst
Comment (Alex-Welsh) [inline doc/source/operations/ceph-management.rst]: There are three sections here all about replacing drives. Could they be merged into one?

File: doc/source/operations/control-plane-operation.rst
Comment (Alex-Welsh) [inline doc/source/operations/control-plane-operation.rst]: This should reference our release train page, because it misses all the steps for syncing pulp content

File: doc/source/operations/control-plane-operation.rst
Comment (Alex-Welsh) [inline doc/source/operations/control-plane-operation.rst]: These instructions are much too simplified. I think it would be best to remove this section and just link to the upgrading.rst page.

File: doc/source/operations/control-plane-operation.rst
Comment (Alex-Welsh) [inline doc/source/operations/control-plane-operation.rst]: Should use `--os-compute-api-version 2.74 --host <hostname>` rather than availability zones

File: doc/source/operations/customising-horizon.rst
Comment (Alex-Welsh) [inline doc/source/operations/customising-horizon.rst]: `*_install_type` is outdated. Can be removed

File: doc/source/operations/customising-horizon.rst
Comment (Alex-Welsh) [inline doc/source/operations/customising-horizon.rst]: Again, outdated. this line can be removed

File: doc/source/operations/openstack-reconfiguration.rst
Comment (Alex-Welsh) [inline doc/source/operations/openstack-reconfiguration.rst]: I think this only applied before pulp and release train. This section can be removed

File: doc/source/operations/wazuh-operation.rst
Comment (Alex-Welsh) [inline doc/source/operations/wazuh-operation.rst:1]: I think this page either needs to focus more on operations (i.e. using wazuh after it's deployed) or just be completely merged into the wazuh.rst doc

Comment (seunghun1ee) [reply doc/source/operations/wazuh-operation.rst:1]: Yes let's merge it to wazuh.rst

File: doc/source/operations/customising-horizon.rst
Comment (seunghun1ee) [reply doc/source/operations/customising-horizon.rst]: I found we don't need to rebuild horizon at all anymore. Deleting the section.

File: doc/source/configuration/wazuh.rst
Comment (Alex-Welsh) [inline doc/source/configuration/wazuh.rst]:
```suggestion
   ``$KAYOBE_CONFIG_PATH/inventory/group_vars/wazuh-manager/wazuh-manager``, in
```

File: doc/source/operations/ceph-management.rst
Comment (Alex-Welsh) [inline doc/source/operations/ceph-management.rst]: nit:
```suggestion
Running Cephadm playbooks
```
or
```suggestion
Running ``cephadm`` playbooks
```

Comment (Alex-Welsh) [reply doc/source/operations/ceph-management.rst]: Same for other uses of "cephadm"

File: doc/source/operations/ceph-management.rst
Comment (Alex-Welsh) [inline doc/source/operations/ceph-management.rst]: The order here has changed a bit since this was written, and there are some new playbooks that run custom commands. The list is here: https://github.com/stackhpc/stackhpc-kayobe-config/blob/stackhpc/2023.1/etc/kayobe/ansible/cephadm.yml

File: doc/source/operations/ceph-management.rst
Comment (Alex-Welsh) [inline doc/source/operations/ceph-management.rst:59]: I think we should get someone with more Ceph experience to review the rest of this file to check it for accuracy.

File: doc/source/operations/control-plane-operation.rst
Comment (Alex-Welsh) [inline doc/source/operations/control-plane-operation.rst]:
```suggestion
* `Back up OpenSearch <https://opensearch.org/docs/latest/tuning-your-cluster/availability-and-recovery/snapshots/snapshot-restore/>`__
```

File: doc/source/operations/control-plane-operation.rst
Comment (Alex-Welsh) [inline doc/source/operations/control-plane-operation.rst]:
```suggestion
The control plane has been configured to collect logs centrally using
OpenSearch, Fluentd, and OpenSearch Dashboards).
```

File: doc/source/operations/control-plane-operation.rst
Comment (Alex-Welsh) [inline doc/source/operations/control-plane-operation.rst]: Should use `systemctl` rather than docker

File: doc/source/operations/control-plane-operation.rst
Comment (Alex-Welsh) [inline doc/source/operations/control-plane-operation.rst]: I'd probably just delete this section. Full version upgrades require a lot more than this. For container tag bumps, we just use service deploy. Either way, I don't think this section is really that useful

File: doc/source/operations/control-plane-operation.rst
Comment (Alex-Welsh) [inline doc/source/operations/control-plane-operation.rst]:
```suggestion
   openstack server create --flavor <flavour name> --network <network name> --key-name <key> --image <Image name> --os-compute-api-version 2.74 --host <hypervisor hostname> <vm name>
```

File: doc/source/operations/control-plane-operation.rst
Comment (Alex-Welsh) [inline doc/source/operations/control-plane-operation.rst]: Patch merged and backported long ago. This can be removed

File: doc/source/operations/customising-horizon.rst
Comment (Alex-Welsh) [inline doc/source/operations/customising-horizon.rst]:
```suggestion
It can be resolved by dropping cached content with ``systemctl restart
kolla-memcached-container.service``. Note this will log out users from Horizon, as Django sessions are
```

**Review by Alex-Welsh — CHANGES_REQUESTED** [review]

File: doc/source/operations/ceph-management.rst
Comment (seunghun1ee) [reply doc/source/operations/ceph-management.rst]: Re-ordered them and added description of new playbooks

File: doc/source/operations/gpu-in-openstack.rst
Comment (seunghun1ee) [inline doc/source/operations/gpu-in-openstack.rst]: Do we still need to specify branch with stackhpc.linux collection?

Comment (MoteHue) [reply doc/source/operations/gpu-in-openstack.rst]: @jovial do you know what the state of vgpu and iommu support is? I see there's a couple of open PRs for this, but none for preemptive/vgpu-iommu

File: doc/source/operations/control-plane-operation.rst
Comment (Alex-Welsh) [inline doc/source/operations/control-plane-operation.rst:43]: Could you add a link in this section to `monitoring.rst` which has more information on how to configure the services?

File: doc/source/operations/control-plane-operation.rst
Comment (Alex-Welsh) [inline doc/source/operations/control-plane-operation.rst]: We should direct users to run the reboot playbook instead, since it covers a few more edge cases than this

File: doc/source/operations/gpu-in-openstack.rst
Comment (Alex-Welsh) [reply doc/source/operations/gpu-in-openstack.rst]: It's already in Kayobe upstream (in antelope) so not required here

Comment (Alex-Welsh) [reply doc/source/operations/gpu-in-openstack.rst]: See https://review.opendev.org/c/openstack/kayobe/+/887200
There's a lot we can remove from this page now, we should instead link to the [upstream docs](https://docs.openstack.org/kayobe/latest/configuration/reference/vgpu.html)

File: doc/source/operations/hardware-inventory-management.rst
Comment (Alex-Welsh) [inline doc/source/operations/hardware-inventory-management.rst:1]: This page is very Bifrost-specific and lots of deployments don't use Bifrost at all. Could you rename it to something like `Bifrost Hardware Inventory Management`?

We'd also have to change the page slightly for example `At its lowest level, hardware inventory is managed in the Bifrost service.` to something like `In most deployments, hardware inventory is managed by the Bifrost service.`

File: doc/source/operations/openstack-reconfiguration.rst
Comment (Alex-Welsh) [inline doc/source/operations/openstack-reconfiguration.rst]: Please also link to the `vault.rst` page in this section

File: doc/source/operations/openstack-reconfiguration.rst
Comment (Alex-Welsh) [inline doc/source/operations/openstack-reconfiguration.rst]: This should use a real [sphinx note](https://sublime-and-sphinx-guide.readthedocs.io/en/latest/notes_warnings.html#notes)

File: doc/source/operations/secret-rotation.rst
Comment (Alex-Welsh) [inline doc/source/operations/secret-rotation.rst]: (``$KOLLA_CONFIG_PATH/passwords.yml`` OR
   ``$KAYOBE_CONFIG_PATH/environments/$KAYOBE_ENVIRONMENT/kolla/passwords.yml``)

**Review by Alex-Welsh — COMMENTED** [review]
Nice work, quite close now. Mainly need to update the vgpu page and get someone with more ceph experince to review the ceph page

**Review by Alex-Welsh — CHANGES_REQUESTED** [review]
Also, the docs job is failing

File: doc/source/operations/control-plane-operation.rst
Comment (seunghun1ee) [reply doc/source/operations/control-plane-operation.rst:43]: ACK

File: doc/source/configuration/vault.rst
Comment (Alex-Welsh) [inline doc/source/configuration/vault.rst]:
```suggestion
This method should only be used for testing. For external TLS on production systems,
```

File: doc/source/operations/bifrost-hardware-inventory-management.rst
Comment (Alex-Welsh) [inline doc/source/operations/bifrost-hardware-inventory-management.rst]: Could you remove the brackets from the references to KAYOBE_CONFIG_PATH (and other env vars) throughout?
```suggestion
file located at ``$KAYOBE_CONFIG_PATH/secrets.yml``.
```

File: doc/source/operations/bifrost-hardware-inventory-management.rst
Comment (Alex-Welsh) [inline doc/source/operations/bifrost-hardware-inventory-management.rst]: Please add a note after this saying that you also need to reconfigure the monitoring services on the controllers, otherwise it will not show up

File: doc/source/operations/bifrost-hardware-inventory-management.rst
Comment (Alex-Welsh) [inline doc/source/operations/bifrost-hardware-inventory-management.rst]: More of those weird brackets here
```suggestion
   cd $KAYOBE_CONFIG_PATH
```

File: doc/source/operations/ceph-management.rst
Comment (Alex-Welsh) [reply doc/source/operations/ceph-management.rst:59]: Still required

File: doc/source/operations/ceph-management.rst
Comment (Alex-Welsh) [inline doc/source/operations/ceph-management.rst]: I may be wrong but I think the spacing is incorrect here
```suggestion
.. note::
```

File: doc/source/operations/control-plane-operation.rst
Comment (Alex-Welsh) [inline doc/source/operations/control-plane-operation.rst]: I've never seen this before. Might be resolved?

File: doc/source/operations/control-plane-operation.rst
Comment (Alex-Welsh) [inline doc/source/operations/control-plane-operation.rst:200]: Controllers before Ceph nodes seems weird, please check this with someone who knows Ceph

File: doc/source/operations/control-plane-operation.rst
Comment (Alex-Welsh) [inline doc/source/operations/control-plane-operation.rst]: The limit on this command is unnecessarily weird and complicated. I'd change it to something a bit easier to understand
```suggestion
   kayobe# kayobe playbook run $KAYOBE_CONFIG_PATH/ansible/reboot.yml --limit compute4,compute5
```

File: doc/source/operations/control-plane-operation.rst
Comment (Alex-Welsh) [inline doc/source/operations/control-plane-operation.rst]: The availability zone trick is the old deprecated way of doing it
```suggestion
you can specify the hypervisor name.
```

File: doc/source/operations/gpu-in-openstack.rst
Comment (Alex-Welsh) [inline doc/source/operations/gpu-in-openstack.rst]: I think there is upstream support for this now, so it's part of host configure, instead of being a separate playbook. This section can be deleted

File: doc/source/operations/migrating-vm.rst
Comment (Alex-Welsh) [inline doc/source/operations/migrating-vm.rst]:
```suggestion
   admin# openstack server migrate --live-migration --block-migration --host hypervisor-01 <VM name or uuid>
```

File: doc/source/operations/openstack-reconfiguration.rst
Comment (Alex-Welsh) [inline doc/source/operations/openstack-reconfiguration.rst]: More weird brackets
```suggestion
``cinder-backup``, edit ``$KAYOBE_CONFIG_PATH/kolla.yml``:
```

File: doc/source/operations/openstack-reconfiguration.rst
Comment (Alex-Welsh) [inline doc/source/operations/openstack-reconfiguration.rst]:
```suggestion
above. Run the following command:
```

File: doc/source/operations/openstack-reconfiguration.rst
Comment (Alex-Welsh) [inline doc/source/operations/openstack-reconfiguration.rst:90]: Please also add a warning here that this is disruptive work

File: doc/source/operations/openstack-reconfiguration.rst
Comment (Alex-Welsh) [inline doc/source/operations/openstack-reconfiguration.rst]: Generally speaking, we should get rid of these admin# and kayobe# prefixes. I think they were used in the admin guide but might render weirdly in the wiki
```suggestion
   openstack compute service set --enable \
```

**Review by Alex-Welsh — DISMISSED** [review]

**Comment by seunghun1ee** [comment]
Too many commits at this point. Let's squash this when merging

**Comment by Alex-Welsh** [comment]
LGTM but someone with ceph knowledge should review the ceph page before we merge

File: doc/source/operations/index.rst
Comment (cityofships) [inline doc/source/operations/index.rst:10]: Shouldn't really be a semi-alphabetical list, if that was the intent

File: doc/source/operations/ceph-management.rst
Comment (cityofships) [reply doc/source/operations/ceph-management.rst]: Indeed, still renders correctly / sphinx does not complain

File: doc/source/operations/ceph-management.rst
Comment (cityofships) [reply doc/source/operations/ceph-management.rst:59]: It's a good start 👍

**Review by cityofships — DISMISSED** [review]
I checked the Ceph related contents - let's merge this

**Comment by Alex-Welsh** [comment]
@seunghun1ee there are still 6 comments left to address on this PR, but they're hidden because there's a lot of threads

**Comment by seunghun1ee** [comment]
> @seunghun1ee there are still 6 comments left to address on this PR, but they're hidden because there's a lot of threads

Oops. Sorry I missed that. I'll address those as well.

File: doc/source/operations/control-plane-operation.rst
Comment (seunghun1ee) [reply doc/source/operations/control-plane-operation.rst]: I agree

File: doc/source/operations/control-plane-operation.rst
Comment (seunghun1ee) [reply doc/source/operations/control-plane-operation.rst]: That is valid syntax and I think it's a necessary example.
Users might want to try rebooting all hypervisors except for a few.
If they have large number of hypervisors it would be painful to write every hypervisors to reboot with comma separated list.

**Review by Alex-Welsh — APPROVED** [review]
Great work Seunghun, now we just need to do it all over again for Caracal :wink:

**Pull request merged**
